packages/anon-creds/src/models/exchange.ts

```diff
@@ -0,0 +1,104 @@
+// TODO: Maybe we can make this a bit more specific?
+export type WalletQuery = Record<string, unknown>
+
+export interface NonRevokedInterval {
+  from?: number | undefined
+  to?: number | undefined
+}
+
+export interface AnonCredsCredentialOffer {
+  schema_id: string
+  cred_def_id: string
+  nonce: string
+  key_correctness_proof: Record<string, unknown>
+}
+
+export interface AnonCredsCredentialRequest {
+  // TODO: Why is this needed? It is just used as context in Ursa, can be any string. Should we remove it?
+  // Should we not make it did related?
+  prover_did: string
+  cred_def_id: string
+  blinded_ms: Record<string, unknown>
+  blinded_ms_correctness_proof: Record<string, unknown>
+  nonce: string
+}
+
+export interface CredValue {
+  raw: string
+  encoded: string // Raw value as number in string
+}
+
+export interface AnonCredsCredential {
+  schema_id: string
+  cred_def_id: string
+  rev_reg_id?: string
+  values: Record<string, CredValue>
+  signature: unknown
+  signature_correctness_proof: unknown
+}
+
+export interface AnonCredsProof {
+  requested_proof: {
+    revealed_attrs: {
+      [key: string]: {
+        sub_proof_index: number
+        raw: string
+        encoded: string
+      }
+    }
```

Comment:
```suggestion
    revealed_attrs: Record<string, {
        sub_proof_index: number
        raw: string
        encoded: string
      }>
    
```

Take it or leave it. Might be a bit more consistent with the other `Record<string, ...>` types.